dra-evolution/pkg/api/claim_types.go

```diff
@@ -1,379 +1,607 @@
 }
 
-// DeviceClaimDetail contains the details of how to fulfill a specific
-// request for devices.
-type DeviceClaimDetail struct {
-	// Class is the name of the DeviceClass to which the requested
-	// devices must belong.
+type ResourceRequestDetail struct {
+	*RequestClassReference `json:",inline,omitempty" protobuf:"bytes,1,opt,name=classReference"`
+
+	// Config defines configuration parameters that apply to the requested resource(s).
+	// They are ignored while allocating the claim.
 	//
-	// +required
-	Class string `json:"class"`
+	// +optional
+	// +listType=atomic
+	Config []ConfigurationParameters `json:"config,omitempty" protobuf:"bytes,1,opt,name=config"`
 
 	// AdminAccess indicates that this is a claim for administrative access
-	// to the devices. Claims with AdminAccess are expected to be used for
+	// to the device(s). Claims with AdminAccess are expected to be used for
 	// monitoring or other management services for a device.  They ignore
 	// all ordinary claims to the device with respect to access modes and
-	// any resource allocations. Ability to create these claims is
-	// controlled via ResourceQuota.
+	// any resource allocations. Ability to request this kind of access is
+	// controlled via ResourceQuota in the resource.k8s.io API.
 	//
-	// Default is false. If true, a DeviceClass must be specified so that
-	// the Driver is known, and only the Constraints fields will be taken
-	// into account for device selection. All devices meeting the
-	// Constraints expressions will be made available as part of the claim
-	// and may be assigned to a container.
+	// Can be combined with a range to ask for access to all devices
+	// on a node which match the requrirements.
 	//
-	// NOTE: This cannot appear in class because the quota code does not do
-	// an indirection. Still searching for a better way to handle this use
-	// case, without creating a new top-level type for it.
+	// Default is false.
 	//
 	// +optional
 	AdminAccess *bool `json:"adminAccess,omitempty"`
 
-	// Constraints is a CEL expression that operates on device attributes.
-	// In order for a device to be considered, this CEL expression and the
-	// Constraints expression from the DeviceClass must both be true.
+	// All these match criteria must be satisfied by all devices allocated for this request.
 	//
 	// +optional
-	Constraints *string `json:"constraints,omitempty"`
-
-	// Device claims may be satisfied by choosing multiple devices instead
-	// of just a single device.
+	// +listType=atomic
+	Match []MatchModel `json:"match,omitempty"`
+
+	// Count defines how many instances are desired. If unset, exactly one
+	// instance must be available. When a range is set, it is possible to
+	// ask for:
+	// - x >= minimum instances up to all that are available
+	// - 0 <= x <= maximum (up to a certain number, with zero instances acceptable)
+	// - minimum <= 0 <= maximum (within a certain range)
```

Comment:
`min <= 0` ?  I think you meant `<= x` ?

Reply:
Yes. But removed, so I need to remember when we bring it back.